src/rgbds.5

```diff
@@ -123,6 +123,33 @@ REPT NumberOfSections
 
     ENDC
 
+ENDR
+
+; Assertions
+
+LONG  NumberOfAssertions
+
+REPT  NumberOfAssertions
+
+  STRING  SourceFile   ; Name of the source file (for printing the failure).
+
+  LONG    Offset       ; Offset into the section where the assertion is located.
+
+  BYTE    Type         ; 0 = Prints the message but allows linking to continue
+                       ; 1 = Prints the message and evaluates other assertions,
+                       ;     but linking fails afterwards
+                       ; 2 = Prints the message and immediately fails linking
+
+  LONG    RPNSize      ; Size of the RPN expression's buffer.
+
+  BYTE    RPN[RPNSize] ; RPN expression, same as patches. Assert fails if == 0.
+
+  LONG    SectionID    ; The section number (of this object file) in which this
+                       ; assert is defined. If it doesn't belong to any specific
+                       ; section (like a constant), this field has the value -1.
+
+  STRING  Message      ; A message displayed when the assert fails, unless empty.
```

Comment:
And how do you know ahead of time if it will be empty? "Unless empty" implies that the field is missing entirely if the message is empty; this is not the same as having just a null terminator to signal emptiness.

Reply:
I understood the "unless empty" as qualifying "displayed", i. e. the message won't be displayed when it's empty (which will format the resulting diagnostic a bit differently).

Reply:
Exactly as Nie said: if the message is a string of length 0, then it's treated as non-existent. (I doubt anyone giving assertions empty messages will be bothered, and if they *really* need the colon, they can just add whitespace.

Reply:
I'm really complaining about the documentation's wording here, not the behavior. The documentation can be misunderstood to mean that the field will be missing instead of an empty string.

Reply:
Then how would you say it?

Reply:
"A message displayed when the assert fails, or an empty string if no message is set"

Reply:
The part that annoys me is that when the user specifies an empty message, it won't be displayed. I would like the documentation to say it (hence my choice of wording), but maybe it's not necessary?

Reply:
"A message displayed when the assertion fails. (If set to the empty string, no message will be shown.)"

Reply:
I didn't like that formulation, but couldn't come up with anything better. Guess I'll roll with this.

Reply:
(If set to the empty string, a generic message will be shown.)

Reply:
No, the generic message ("Assertion failure") is always shown, to explicit the source of the error message. Though, it might actually be unnecessary, since the line number is printed, and can be traced back... thoughts?

Reply:
In ca65, for comparison, this:
```
.assert 0, error, "nope"
```
produces this:
```
assert.s(1): Error: nope
```
And this:
```
.assert 0, error
```
produces this:
```
assert.s(1): Error: Assertion failed
```
There's a distinction between a missing argument and an empty string, as this:
```
.assert 0, error, ""
```
produces this:
```
assert.s(1): Error: 
```

Reply:
The ca65 behavior seems sensible, although the empty string case is mostly irrelevant and you can leave it out if it is problematic.

Reply:
Implemented Nie's suggestion, both in the documentation and in actual behavior.